### ISSUE TITLE
UI Style reuse: Inheritance + assets

### ISSUE THREAD

**Issue by Weibye** [bevyengine/bevy#5516]
## What problem does this solve or what need does it fill?

At the moment `Style` is incredibly verbose to use, just have a look at the [ui example](https://github.com/bevyengine/bevy/blob/main/examples/ui/ui.rs). 
In order to improve ergonomics and reduce boilerplate when dealing with UI we need to do something about it and we should strive to achieve the level of UX that we see in other parts of the engine.

## What solution would you like?

In order to do so we are going to need two related but IMO orthogonal features:

1. Style inheritance
2. Style as assets

I don't know exactly what these would look like in implementation, but I'm hoping some of you have good ideas for design-approaches that we then split out into separate issues or PRs.

**This issue should be about figuring out a scope and requirements for both of these features**

## What alternative(s) have you considered?

- Leave style as is and keep verbose + cumbersome UX
- There might also be design-approaches for style reuse that solves the problem without using inheritance but I don't know enough about this domain, so please share your 🧠 

## Additional context

We might find that we are blocked on other features like scenes or reflection on some of this but in that case it is good to know exactly where the blocker is so that we can push to unblock those 😄

**Comment by alice-i-cecile** [comment]
Style inheritance is going to be extremely controversial FYI. Personally, I'm hopeful for a compositional design, as laid out in https://github.com/bevyengine/rfcs/pull/1

**Comment by Weibye** [comment]
> Style inheritance is going to be extremely controversial FYI. Personally, I'm hopeful for a compositional design, as laid out in [bevyengine/rfcs#1](https://github.com/bevyengine/rfcs/pull/1)

Good point, I've updated the issue to better reflect weighing the problem in this direction

**Comment by IceSentry** [comment]
I think inheritance would be fine and not too controversial if it was opt in
